R/NDVI.R

```diff
@@ -0,0 +1,57 @@
+#' @importFrom magrittr "%>%"
+
+
+#' @title NDVI by calendar month or lunar month
+#'
+#' @description Summarize NDVI data to monthly or lunar monthly level
+#'
+#' @param level specify 'monthly' or 'newmoon'
+#' @param path specify where to locate Portal data
+#'
+#' @export
+#'
+summarize_ndvi <- function(level = "monthly", path = '~') {
+  NDVI=read.csv(FullPath('PortalData/NDVI/monthly_NDVI.csv', path), na.strings=c("-99"), stringsAsFactors = FALSE)
+  moon_dates <- read.csv(FullPath('PortalData/Rodents/moon_dates.csv', path), na.strings = c(""), stringsAsFactors = FALSE)
+
+  if(!all(c("year", "month") %in% names(NDVI))){
+    NDVI$month <- lubridate::month(paste0(NDVI$date, "-01"))
+    NDVI$year <- lubridate::year(paste0(NDVI$date, "-01"))
+    NDVI$ndvi <- NDVI$NDVI
+  }
+  if(!"date" %in% names(NDVI)){
+    NDVI$date <- as.Date(paste(NDVI$year, NDVI$month, "01", sep = "-"))
+  }
+  NDVI$ndvi[NDVI$ndvi == "NA"] <- NA
```

Comment:
And having to deal with NAs here?

Reply:
ditto

Reply:
Okay, this should no longer be necessary then?

Reply:
looks like that's the case. i'll remove it